src/software/thunderscope/common/build_parameter_tree.py

```diff
@@ -0,0 +1,214 @@
+    # in the dictionary.
+    options = MessageToDict(descriptor.GetOptions(), preserving_proto_field_name=True)
+
+    try:
+        min_max = options["[TbotsProto.bounds]"]
+    except KeyError:
+        raise KeyError("{} missing ParameterRangeOptions".format(key))
+
+    return {
+        "name": key,
+        "type": "slider",
+        "value": value,
+        "default": value,
+        "limits": (min_max["min_double_value"], min_max["max_double_value"],),
+        "step": 0.01,
+    }
+
+
+def __create_enum_parameter(key, value, descriptor):
+    """Converts an enum field in a protobuf to a ListParameter. Uses
+    the options to lookup all possible enum values and provides them
+    as a dropdown option.
+
+    :param key: The name of the parameter
+    :param value: The default value
+    :param descriptor: The proto descriptor
+
+    """
+    options = []
+
+    for enum_desc in descriptor.enum_type.values:
+        options.append(enum_desc.name)
+
+    # The list index is indexed from 1
+    current_enum_index = value - 1
+
+    return parametertree.parameterTypes.ListParameter(
+        name=key,
+        default=None,
+        value=descriptor.enum_type.values[current_enum_index].name,
+        limits=options,
+    )
+
+
+def __create_bool_parameter(key, value, _):
+    """Convert a bool field in proto to a BoolParameter
+
+    :param key: The name of the parameter
+    :param value: The default value
+    :param _: The proto descriptor, unused for bool
+
+    """
+    return {"name": key, "type": "bool", "value": value}
+
+
+def __create_string_parameter_writable(key, value, descriptor):
+    """Convert a string field in proto to a StrParameter
+
+    :param key: The name of the parameter
+    :param value: The default value
+    :param descriptor: The proto descriptor
+
+    """
+    return {"name": key, "type": "text", "value": " "}
+
+
+def __create_parameter_read_only(key, value, descriptor):
+    """Convert a string field in proto to a read only Str Parameter
+
+    :param key: The name of the parameter
+    :param value: The default value
+    :param descriptor: The proto descriptor
+
+    """
+    return {"name": key, "type": "str", "value": value, "readonly": True}
+
+
+def config_proto_to_field_list(
+    message, read_only=True, search_term=None, search_filter_threshold=60
+):
+    """Converts a protobuf to a pyqtgraph parameter tree dictionary
+    that can loaded directly into a ParameterTree
+
+    https://pyqtgraph.readthedocs.io/en/latest/parametertree/index.html
+
+    :param message: The message to convert to a dictionary
+    :param read_only: Whether the parameters should be read only or writable
+    :param search_term: The search filter
+    :param search_filter_threshold: the search filter threshold
+
+    """
+    field_list = []
+
+    for descriptor in message.DESCRIPTOR.fields:
+
+        key = descriptor.name
+        value = getattr(message, descriptor.name)
```

Comment:
Just tried hard coding the value for the enum
```py
if key == 'error_code':
    value = ErrorCode.LOW_CAP
```
 and it seems like it just prints the integer associated with the value, rather than the string representation of it. Having a look at the implementation of `__create_enum_parameter` could help.